straxbra/plugins.py

```diff
@@ -402,21 +428,42 @@ def compute(self, peaks):
         r = np.full_like(p, np.nan, dtype=self.dtype)
 
         if len(p) >= 0:
-            r = self.weighted_sum(p, self.pmt_positions, r)
+            for i in range(0, len(p)):
+                r[i] = self.reconstructed_position(p[i])
         else:
             return dict(x=np.zeros(0, dtype=np.float32),
                         y=np.zeros(0, dtype=np.float32))
         if not r.flags['C_CONTIGUOUS']:
             r = r.copy(order='C')
         return r
 
+    # Function to return non-negative value corresponding to (radial position - radius TPC) if inside TPC,
+    # used for constraints
+    @staticmethod
+    def insidevolume(inputs):
+        p_TPC_radius = 35
```

Comment:
It would be nice if this could be an option or a class variable, but that might be tricky in this scope.

Reply:
Since this is really just the first draft for the LRF position reconstruction solely fitted to our current XeBRA setup I would leave it like that for the moment but will keep it in mind for further extensions in future versions.